packages/output/src/browser/output-contribution.ts

```diff
@@ -122,19 +122,29 @@ export class OutputContribution extends AbstractViewContribution<OutputWidget> i
     registerCommands(registry: CommandRegistry): void {
         super.registerCommands(registry);
         registry.registerCommand(OutputCommands.CLEAR__WIDGET, {
-            isEnabled: widget => this.withWidget(widget),
-            isVisible: widget => this.withWidget(widget),
-            execute: () => this.widget.then(widget => widget.clear())
```

Comment:
as before it was good, except `execute` should use `withWidget` as well to execute a command exactly on the same widget as `isEnabled` and `isVisible`.

The same for other command handlers.